src/main/java/uk/gov/register/db/mappers/RecordMapper.java

```diff
@@ -1,26 +1,42 @@
 package uk.gov.register.db.mappers;
 
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.dropwizard.jackson.Jackson;
 import org.skife.jdbi.v2.StatementContext;
 import org.skife.jdbi.v2.tweak.ResultSetMapper;
+import uk.gov.register.core.Item;
 import uk.gov.register.core.Record;
 
+import java.io.IOException;
 import java.sql.ResultSet;
 import java.sql.SQLException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
 
 public class RecordMapper implements ResultSetMapper<Record> {
     private final EntryMapper entryMapper;
-    private final ItemMapper itemMapper;
+    private final ObjectMapper objectMapper;
 
     public RecordMapper() {
         this.entryMapper = new EntryMapper();
-        this.itemMapper = new ItemMapper();
+        objectMapper = Jackson.newObjectMapper();
     }
 
     @Override
     public Record map(int index, ResultSet r, StatementContext ctx) throws SQLException {
+        List<Item> items = Arrays.asList((String[]) r.getArray("content").getArray()).stream().map(c -> {
+            try {
+                return new Item(objectMapper.readValue(c, JsonNode.class));
```

Comment:
Judging by the documentation (https://www.postgresql.org/docs/9.5/static/functions-aggregate.html) the `sha256hex` array and the `content` array should remain in the same order they were read from the joined tables. I think it's therefore safe to loop over both arrays and use the `Item(hash, content)` contructor, rather than recomputing the hash.